### PR TITLE
define acceptable node versions, and enforce at install time

### PR DESCRIPTION
**Pull request by pes10k**
Simple change that 

1. defines allowed node versions in `package.json`
2. enforce this in`.npmrc`

Resolves brave/brave-browser#42911

## Submitter Checklist:

- [X] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [X] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [X] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [X] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [X] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [X] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [X] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A
  - [ ] https://github.com/brave/brave-browser/wiki/Brave-Wallet

### CODE REVIEW

**Comment by fallaciousreasoning** [comment]
Probably doesn't hurt having this but the recommended build instructions are to checkout `brave-core` now, so this should be caught during init

Additionally, we should be careful not to let these get out of sync :laughing:

**Comment by pes10k** [comment]
ah, gotcha gotcha! Two suggestions then

1. maybe it'd be good to also force the version check with a .npmrc in brave-core then too?
2. making sure things don't fall out of sync sounds very good ha :) the wiki/README/other documentation already needs to stay in sync with brave-core's package.json though. I dont know where the "one true" definition should be but, just noting that things already need to stay in sync, and so maybe it can all be hoisted into one central location? WDYT?

File: package-lock.json [hunk 1]
```diff
@@ -6,10 +6,14 @@
   "packages": {
     "": {
       "name": "brave",
-      "version": "1.75.0",
+      "version": "1.75.100",
```
[the remark this answers is not in the document]
Comment (pes10k) [reply package-lock.json]: ah, not sure what happened there, but fixed!

**Comment by pes10k** [comment]
@goodov just pinging on this, is there anything further needed from me on this?

**Comment by goodov** [comment]
> @goodov just pinging on this, is there anything further needed from me on this?

I don't think so. feel free to merge!

**Comment by pes10k** [comment]
thanks! but no can do yet, i think i still need another approval from @brave/devops-team 🤔

**Review by mihaiplesa — APPROVED** [review]

**Comment by pes10k** [comment]
thanks to ya both!

**Comment by reneleonhardt** [comment]
@pes10k I can't find a build workflow.
node 24 LTS is 3 months away, is Brave already compatible with the excluded node 22?

**Comment by goodov** [comment]
@mihaiplesa this needs to be synced with brave-core

**Pull request merged**